### DISCUSSION TITLE
Use websockets on the long term

### DISCUSSION THREAD

**Discussion by nerdoc** [django-commons/django-unicorn#224] [Ideas]
Maybe it's blasphemic, and I don't want to start a troll question... But: I used a bunch of similar projects (Sockpuppet, reactor, partly htmlx), and of course tested Vue, Svelte, Alpine etc.
I think Unicorn is the most convenient and most pythonic/djangonic way to get that stuff done. It just comes to my mind that ajax calls are slower than websockets, because you have to make a HTTP request each time, which costs time, as said in [another discussion](https://github.com/adamghill/django-unicorn/discussions/171).

The main issue IMHO is another: websockets allow bi-directional communication, so, using channels, push notifications could be established for a component.

Is there something planned for Unicorn as well? Is there a possibility Unicorn moves to websockets any time in future?

**Comment by adamghill** [comment]
@nerdoc Blasphemy! ;) Not dealing with setting up django-channels, redis, etc and a simpler deployment seems like an ok trade-off to me for the sorts of sites that I end up building.

I'm not completely against websockets, though. If someone wanted to create a another transport for `Unicorn` I'd help by providing extension points, hooks, etc. But, it's probably not something I would build by myself -- there are too many other features that I personally want to build into `Unicorn` instead. 👍

**Reply by nerdoc** [reply]
Ok, channels, redis are less complicated to setup than the whole bunch of Javascript when you need to build a bigger project...

But my main concern is the "server push" feature. Is there any way you can think of to make unicorn "reactive" to server-side messages? Because that's what is the killer feature of modern frameworks: updating the UI from server side.

**Reply by adamghill** [reply]
The only ways I know of are SSE (https://github.com/fanout/django-eventstream) or websockets.

**Reply by nerdoc** [reply]
Oh, thanks, didn't know SSE/django-eventstream!
